ftl/python/builder.py

```diff
@@ -36,11 +36,17 @@
 class Python(builder.JustApp):
     def __init__(self, ctx):
         self.descriptor_files = [_REQUIREMENTS_TXT]
+        # TODO(aaron-prindle) map  python_version to interpreter_parser
+        # will prob need to pass in args to the builder
+        self.interpreter_parser = lambda: 'python2.7'
+        self.pkg_txt_parser = ftl_util.descriptor_parser  # detect phase here
```

Comment:
Not sure I understand this part. Maybe some comments would help?

Reply:
The parsers are used to generate the cache keys (and args) for each Layer Creation Step.  
interpreter_parser returns the interpreter layer cache key
pkg_txt_parser returns the pkg_txt value to pass to the dependency layer function(s)
pkg_key_parser returns the dependency layer cache key
pkg_txt and pkg_key are different as for some values you can on something different then what you want to pass to the pkg manager. 

I am going to change this to introduce a Layer, then the builder returns a list of layers and the builder runner executes and compiles the layers